src/guide/configuring-labs.md

```diff
@@ -0,0 +1,117 @@
+
+<p class="tip">A lab must have an `ml.yaml` file, otherwise it won't be executable.</p>
+
+A configuration setting usually comes with a name followed by one or more values. For example, if we want to configure a environment in which our lab is executed, we can use MachineLabs' `dockerImageId` property with a dedicated value:
+
+```yaml
+dockerImageId: keras_v2-0-x_python_3-1
+```
+
+If a configuration property can have multiple values, we specificy this as a list like this:
+
+```yaml
+inputs:
+  - name: some-value
+    url: https://some-url.com
+```
+
+In this case we're configuring a [lab input](#lab-inputs), which can be a dataset that will be fetched from a specified `url`. Notice that `inputs` is the configuration property while `name` and `url` a part of a single configuration item, which is part of a list of values.
+
+Let's take a look at what configuration options are available in MachineLabs.
+
+## Lab environment
+
+Every lab we execute will be executed in a certain environment. The environment describes things like what programming language and libraries will be available at execution time.
+
+MachineLabs uses [docker](https://docker.com) containers to run labs in isolated and reproducable environments. It comes with a set of prebuilt images that we can use to execute our code. To specify a lab's environment, we use the `dockerImageId` configuration property.
+
+```yaml
+dockerImageId: docker_image_name
+```
+
+The following docker images are supported (more will be added in the near future):
+
+- **keras_v2-0-x_python_3-1** - Keras 2.0.x and Python 3.1
+- **keras_v2-0-x_python_2-1** - Keras 2.0.x and Python 2.1
+
+This means, if we want to run our lab with Keras version 2.0.x and Python version 3.1, our configuration would look like this:
+
+```yaml
+dockerImageId: keras_v2-0-x_python_3-1
+```
+
+Configuring custom docker images is a feature that is planned for future versions.
+
+## Lab inputs (datasets)
+
+Obviously there's no Machine Learning without data. Labs need access to datasets and one way to get hold of them is using **inputs**. Inputs are basically metadata that describe what data needs to be downloaded before a lab is executed.
+
+An input has a `url`, which is the endpoint from which that data should be downloaded from, and a `name`, which specifies under what name the data is stored on the file system in a special `inputs` directory, so it can be later accessed by the lab. MachineLabs has full internet access so we can basically download from any reachable place.
+
+<p class="tip">All inputs are downloaded in a special `inputs` directory and can be accessed from within our labs using common file system operations.</p>
+
+For example, the following configuration sets up an input that fetches the famous MNIST dataset and saves it as `mnist.npz` into the `inputs `directory.
```

Comment:
I think we should somewhere point out what we also pointed out in the quickstart<

>The name that we specify for the input doesn’t have to match the actual download name. This enables us to give different names to files that would otherwise conflict with their names. However, it is mandatory to specify a name for each input even if we don’t need a different name.

Otherwise, it may not be obvious that the `name` is mandatory and that one can use it to resolve conflicting names.